src/panels/lovelace/components/hui-code-editor.ts

```diff
@@ -0,0 +1,69 @@
+import CodeMirror from "codemirror";
+import "codemirror/mode/yaml/yaml";
+// tslint:disable-next-line
+import codeMirrorCSS from "codemirror/lib/codemirror.css";
+import { fireEvent } from "../../../common/dom/fire_event";
+
+let _this;
+
+declare global {
+  interface HASSDomEvents {
+    "code-changed": {
+      value: string;
+    };
+  }
+}
+
+export class HuiCodeEditor extends HTMLElement {
```

Comment:
Let's not call it a code editor if we hardcode it for yaml ? Let's call it a YAML editor